chain/client/src/client.rs

```diff
@@ -97,15 +103,27 @@ impl Client {
         let num_block_producer_seats = config.num_block_producer_seats as usize;
         let data_parts = runtime_adapter.num_data_parts();
         let parity_parts = runtime_adapter.num_total_parts() - data_parts;
+
+        let doomslug = Doomslug::new(
+            block_producer.as_ref().map(|x| x.account_id.clone()),
```

Comment:
may be let's move BlockProducer to primitives (and rename  to BlockSigner) and use it in Doomslug instead of splitting here?